### PR TITLE
feat: Encoding API Improvements

### PR DESCRIPTION
**Pull request by nabetti1720**
### Description of changes

To improve compatibility with encodings supported by Node.js, the following actions have been taken.
https://nodejs.org/api/util.html#whatwg-supported-encodings

- For encodings supported by LLRT, modified so that they can also be specified by aliases.
- Reduced excessive normalization and made label judgments a little stricter.
- Since `iso-8859-1` was an alias, it was managed by `windows-1252`.
- `ENCODING_MAP` was introduced with the expectation of improved maintenance ~and lookup performance~.

### Checklist

- [x] Created unit tests in `tests/unit` and/or in Rust for my feature if needed
- [x] Ran `make fix` to format JS and apply Clippy auto fixes
- [x] Made sure my code didn't add any additional warnings: `make check`
- [x] Updated documentation if needed ([API.md](API.md)/[README.md](README.md)/Other)

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

File: llrt_utils/src/encoding.rs [hunk 1]
```diff
@@ -2,62 +2,67 @@
...
-    pub enum Encoder {
-        Hex,
-        Base64,
-        Utf8,
-        Iso88591,
-        Utf16le,
-    }
-}
+static ENCODING_MAP: Lazy<HashMap<&'static str, Encoder>> = Lazy::new(|| {
+    let mut map = HashMap::new();
```
Comment (richarddavison) [inline llrt_utils/src/encoding.rs:18]: We should create with capacity as we know up front how many items to put in there

**Review by richarddavison — APPROVED** [review]
Hi @nabetti1720! I like the aliasing and more simple approach. Not sure if this yields much performance benefit (except for maybe one less string allocation to normalize the key) but it's more maintainable for sore.

**Comment by nabetti1720** [comment]
Hi @richarddavison .
Yes. In the previous implementation, it was necessary to create a new enum when dealing with aliases that exceeded the normalization range (e.g., treating `unicode-1-1-utf8` as `utf-8`), and this proposal was made to avoid this.

We believe that by minimizing the normalization, we can now evaluate with the same string as the actual label. 

Finally, as for performance, this level of performance would not make a difference. :)

File: types/buffer.d.ts [hunk 2]
```diff
@@ -1,12 +1,9 @@
...
   export type BufferEncoding =
-    | "utf8"
     | "utf-8"
-    | "utf16le"
     | "utf-16le"
+    | "windows-1252"
     | "base64"
-    | "iso88591"
-    | "iso-8859-1"
     | "hex";
```
Comment (Sytten) [inline types/buffer.d.ts:7]: We might want to keep the typing for the variations

Comment (nabetti1720) [reply types/buffer.d.ts:7]: It is not accurate to leave only some aliases. We now enumerate all labels that can be mapped.

**Review by richarddavison — APPROVED** [review]

**Pull request merged**